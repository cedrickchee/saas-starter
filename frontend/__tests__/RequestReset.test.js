import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import RequestReset, {
  REQUEST_RESET_MUTATION
} from '../components/RequestReset';

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: 'cedrick.chee@gmail.com' }
    },
    result: {
      data: { requestReset: { message: 'success', __typename: 'Message' } }
    }
  }
];

describe('<RequestReset/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it('calls the mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    // simulate typing an email
    wrapper.find('input').simulate('change', {
      target: { name: 'email', value: 'cedrick.chee@gmail.com' }
    });
    wrapper.find('form').simulate('submit');
    await wait(2); // brittle! why do I have to wait 2 ms, not 0?
    wrapper.update();
    expect(wrapper.find('p').text()).toContain(
      'Success! Check your email for a reset link!'
    );
  });
});
