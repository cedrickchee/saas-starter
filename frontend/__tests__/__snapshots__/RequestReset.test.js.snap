// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `
<form
  className="Form-i6nmqi-0 geiMmO"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Request a password reset
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset
    </button>
  </fieldset>
</form>
`;
